### PR TITLE
GH-672: Update SMS recipient

### PR DESCRIPTION
**Pull request by jamesmontemagno**
### Description of Change ###
Adds a constructor for a single recipient.

### Bugs Fixed ###

- Related to issue #672

Provide links to issues here. Ensure that a GitHub issue was created for your feature or bug fix before sending PR.

### API Changes ###

List all API changes here (or just put None), example:

Added: 
 
- `SmsMessage.ctr(string, string);`

### Behavioral Changes ###

None

### PR Checklist ###

- [x] Has tests (if omitted, state reason in description)
- [x] Has samples (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Updated documentation ([see walkthrough](https://github.com/xamarin/Essentials/wiki/Documenting-your-code-with-mdoc))

### CODE REVIEW

File: Xamarin.Essentials/Sms/Sms.shared.cs [hunk 1]
```diff
@@ -31,6 +31,11 @@ public SmsMessage()
         {
         }
 
+        public SmsMessage(string body, string recipient)
```
Comment (Mrnikbobjeff) [inline Xamarin.Essentials/Sms/Sms.shared.cs]: May I suggest outsourcing the body setting to a private constructor? this way we can elide setting the recipient list by allocating an array and copy constructing a list as well as allocating linq decorators. 

```
    public SmsMessage(string body, string recipient)
            : this(body)
        {
            Recipients = string.IsNullOrWhiteSpace(body) ? new List<string>(0) : new List<string>() {recipient};
        }

        private SmsMessage(string body)
        {
            Body = body;
        }

        public SmsMessage(string body, IEnumerable<string> recipients)
         : this(body)
        {
            Recipients = recipients?.Where(x => !string.IsNullOrWhiteSpace(x)).ToList();
        }`

Comment (mattleibow) [reply Xamarin.Essentials/Sms/Sms.shared.cs]: Good point. I would even say go further and forget about the chaining of constructors:

```csharp
public SmsMessage() {
}
public SmsMessage(string body, string recipient) {
    Body = body;
    if (!string.IsNullOrWhiteSpace(recipient))
        Recipients.Add(recipient);
}
public SmsMessage(string body, IEnumerable<string> recipients) {
    Body = body;
    if (recipients != null) {
        Recipients.AddRange(recipients.Where(x => !string.IsNullOrWhiteSpace(x)));
    }
}
// . . .
public List<string> Recipients { get; set; } = new List<string>();
```

We don't have to care about the initial size of the list as the default size is 4 (if corefx is anything to go by).

**Comment by Mrnikbobjeff** [comment]
Well I believe if I were to add multiple recipients you would use the constructor accepting an IEnumerable. If you use the constructor with a single recipient I would rather explicitly set the Capacity of the List as this will probably be enough for the common use case of this constructor.

**Comment by jamesmontemagno** [comment]
Updated! @mattleibow

File: docs/en/FrameworksIndex/xamarin-essentials-android.xml [hunk 2]
```diff
@@ -10,22 +10,18 @@
       <Member Id="P:Xamarin.Essentials.Accelerometer.IsMonitoring" />
     </Type>
     <Type Name="Xamarin.Essentials.AccelerometerChangedEventArgs" Id="T:Xamarin.Essentials.AccelerometerChangedEventArgs">
-      <Member Id="M:Xamarin.Essentials.AccelerometerChangedEventArgs.#ctor(Xamarin.Essentials.AccelerometerData)" />
```
Comment (mattleibow) [inline docs/en/FrameworksIndex/xamarin-essentials-android.xml]: I think something went wrong with this file... There are too many changes.

**Comment by jamesmontemagno** [comment]
@mattleibow should be good now!

**Pull request merged**